EssentialsSpawn/src/main/java/com/earth2me/essentials/spawn/EssentialsSpawnPlayerListener.java

```diff
@@ -40,21 +40,23 @@ void onPlayerRespawn(final PlayerRespawnEvent event) {
             return;
         }
 
-        if (VersionUtil.getServerBukkitVersion().isHigherThanOrEqualTo(VersionUtil.v1_16_1_R01) && event.isAnchorSpawn() && ess.getSettings().isRespawnAtAnchor()) {
-            return;
-        }
-
         if (ess.getSettings().getRespawnAtHome()) {
             final Location home;
 
-            Location bed = null;
-            if (ess.getSettings().isRespawnAtBed()) {
+            Location respawnLocation = null;
+            if (ess.getSettings().isRespawnAtBed() &&
+                    (!VersionUtil.getServerBukkitVersion().isHigherThanOrEqualTo(VersionUtil.v1_16_1_R01) ||
+                    (!event.isAnchorSpawn() || ess.getSettings().isRespawnAtAnchor()))) {
                 // cannot nuke this sync load due to the event being sync so it would hand either way
-                bed = user.getBase().getBedSpawnLocation();
+                try {
+                    respawnLocation = user.getBase().getRespawnLocation();
+                } catch (NoSuchMethodError e) { // For versions prior to 1.16.
+                    respawnLocation = user.getBase().getBedSpawnLocation();
+                }
```

Comment:
Please use `VersionUtil` checks here, that stack trace can be expensive especially if this is hot code